### ISSUE TITLE
Support APCA/WCAG 3

### ISSUE THREAD

**Issue by darekkay** [darekkay/a11y-contrast#4]
WCAG 3 will change how contrast ratios are calculated. The specs are still under development, but `a11y-contrast` could already implement the current status.

- https://cliambrown.com/contrast/
- https://www.myndex.com/APCA/
- https://yatil.net/blog/wcag-3-is-not-ready-yet
- [APCA is no longer part of WCAG3](https://toot.cafe/@aardrian/113801092971854851)

Note that the spec is not stable yet, which is [reflected in the license](https://news.ycombinator.com/item?id=31422247).

**Comment by Myndex** [comment]
> _WCAG 3 will change how contrast ratios are calculated. The specs are still under development, but `a11y-contrast` could already implement the current status._

Hi @darekkay 

Indeed, APCA is in public beta and being adopted and integrated in more tools. Most tools have both APCA (the candidate for WCAG 3)  and the existing WCAG 2, so comparisons can be made. It's important to note that APCA is not backwards compatible to WCAG 2, which i will explain in a moment.

Because of concerns regarding backwards compatibility, we've also released **Bridge-PCA**, which is completely backwards compatible with WCAG 2, but provides substantially improved readability. There's a simple demo tool with further information at https://www.myndex.com.BPCA/

Both are available as NPM packages:

    npm i apca-w3

    npm i bridge-pca

### Regarding backwards compatibility
This recent article demonstrates how [WCAG 2 contrast in incorrect ](https://atangledwebweweave.com/whats-red-black-also-not-read-573b9c0a97ed)and detrimental to color vision deficiency. I have a list of other articles and resources on my profile page here: http://git.myndex.com

APCA corrects this by using modern perceptually uniform techniques, based on actual human perception. You can see this  per these charts and comparison table. In the following table (click it to see it at 100%) as color get darker, the WCAG 2 contrast levels get harder and harder to read, but APCA maintains constant contrast across the visible range.

_Click To Enlarge_
<img width="480" alt="ColumnCompareAll400" src="https://user-images.githubusercontent.com/42009457/148695645-02b8a443-00b8-4633-8939-8ed7d3edfde2.png">

These charts below were created from the data in the above examples. In the charts below:
- The White area is where both WCAG 2 and APCA reject the same colors.
- The Purple area is where they both _PASS_ the sam colors
- The Cyan blue area is where WCAG 2 passes but APCA rejects the colors
- The Magenta area is where APCA passes colors that WCAG 2 (incorrectly) rejects.

<img width="200" alt="W31textLc45" src="https://user-images.githubusercontent.com/42009457/148695327-1b2b792c-a530-4534-af06-fbc705f7cc90.jpg"> <img width="200"  alt="W451textLc60" src="https://user-images.githubusercontent.com/42009457/148695330-68e72faf-7340-4f17-a96a-3732d3517725.jpg"> <img width="200"  alt="W71textLc75" src="https://user-images.githubusercontent.com/42009457/148695332-3cbdc62a-7de8-4bb0-9f5a-57c99488f29b.jpg"> <img width="200"  alt="W71textLc90" src="https://user-images.githubusercontent.com/42009457/148695333-9f27dbfe-6d2c-4bc8-80b0-3eb59219ceb8.jpg">
### Legend
<img width="175" alt="chart info and legend" src="https://user-images.githubusercontent.com/42009457/148695334-f8f188ef-815b-4f87-bf6d-dcc6f078828a.png">

It's because of the WCAG 2 incorrect rejections of light colors that APCA is not backwards compatible, But as this comparison shows, APCA is significantly better at preserving readability than WCAG 2.

But Bridge-PCA is backwards compatible — and to get that, what is lost is some design flexibility. But APCA and Bridge-PCA both fix the notorious "dark color problem" of WCAG 2.

Please feel free to ask questions or comment. The [discussions tab is open](https://github.com/Myndex/SAPC-APCA/discussions) at the repo.


Thank you

Andy
